packages/ui/src/button/Button.tsx

```diff
@@ -1,133 +1,74 @@
+  warning?: boolean
+  loading?: boolean
+  size?: ButtonSize
+  leftIcon?: React.ReactNode
+  rightIcon?: React.ReactNode
+  onClick?: React.MouseEventHandler<HTMLButtonElement>
 } & React.ButtonHTMLAttributes<HTMLButtonElement>
 
-const backgroundColors = {
-  green: 'bg-green',
-  red: 'bg-red',
-  blue: 'bg-blue',
-  violet: 'bg-violet',
-  lightGreen: 'bg-green-light',
-  lightRed: 'bg-red-light',
-  lightBlue: 'bg-blue-light',
-  lightViolet: 'bg-violet-light',
-  darkGreen: 'bg-green-dark',
-  darkRed: 'bg-red-dark',
-  darkBlue: 'bg-blue-dark',
-  darkViolet: 'bg-violet-dark',
-}
-
-const borderColors = {
-  green: 'border-green',
-  red: 'border-red',
-  blue: 'border-blue',
-  violet: 'border-violet',
-  lightGreen: 'border-green-light',
-  lightRed: 'border-red-light',
-  lightBlue: 'border-blue-light',
-  lightViolet: 'border-violet-light',
-  darkGreen: 'border-green-dark',
-  darkRed: 'border-red-dark',
-  darkBlue: 'border-blue-dark',
-  darkViolet: 'border-violet-dark',
-}
-
-const textColors = {
-  green: 'text-green',
-  red: 'text-red',
-  blue: 'text-blue',
-  violet: 'text-violet',
-  lightGreen: 'text-green-light',
-  lightRed: 'text-red-light',
-  lightBlue: 'text-blue-light',
-  lightViolet: 'text-violet-light',
-  darkGreen: 'text-green-dark',
-  darkRed: 'text-red-dark',
-  darkBlue: 'text-blue-dark',
-  darkViolet: 'text-violet-dark',
-}
-
 const Button = ({
   text,
-  variant,
-  disabled,
-  loading,
-  size,
-  color,
+  variant = 'primary',
+  disabled = false,
+  warning = false,
+  loading = false,
+  size = 'M',
   leftIcon,
   rightIcon,
-  style,
   onClick,
   className,
   children,
   ...rest
 }: Props) => {
   const sizeToClass = {
-    S: 'rounded-[16px] w-[100%] h-[32px] box-border',
-    M: 'rounded-[24px] w-[100%] h-[48px] box-border',
-    L: 'rounded-[26px] w-[100%] h-[52px] box-border',
+    S: 'rounded-[16px] h-[32px] note-3',
+    M: 'rounded-[24px] h-[48px] body-1',
+    L: 'rounded-[26px] h-[52px] title-1',
   }
+
   const variantToClass = {
-    primary: `w-[100%] text-white box-border ${backgroundColors[color]}`,
-    outline: `w-[100%] ${textColors[color]} border ${borderColors[color]} bg-transparent`,
+    primary: `text-white`,
+    outline: `border bg-transparent`,
   }
 
   const disabledStyleVariantToClass = {
     primary: `disabled:bg-gray-200`,
     outline: `disabled:text-gray-200 disabled:border-gray-200`,
   }
 
-  const textStyles = () => {
-    switch (size) {
-      case 'L':
-        return 'button-1'
-      case 'M':
-        return 'button-2'
-      case 'S':
-        return 'button-3'
-      default:
-        return 'button-1'
-    }
+  const warningStyleVariantToClass = {
+    primary: `bg-red-dark`,
+    outline: `border-red-dark text-red-dark`,
   }
-
   return (
     <button
-      className={cn(sizeToClass[size], variantToClass[variant], disabled && disabledStyleVariantToClass[variant], className)}
+      className={cn(
+        'w-full',
+        'px-[24px]',
```

Comment:
고정 스타일은 한줄로 써도 좋아보이네요 :)
```suggestion
        'w-full px-[24px]',
```